### PR TITLE
OnError callback function for federated updater

### PR DESCRIPTION
**Pull request by mwielgus**
The handler can be useful for precise logging and events recording. Update returns just a single, first received error.

cc: @quinton-hoole @wojtek-t @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31641)

### CODE REVIEW

File: federation/pkg/federation-controller/util/federated_updater.go [hunk 1]
```diff
@@ -69,7 +69,11 @@ func NewFederatedUpdater(federation FederationView, add, update, del FederatedOp
 	}
 }
 
-func (fu *federatedUpdaterImpl) Update(ops []FederatedOperation, timeout time.Duration) error {
+func (fu *federatedUpdaterImpl) Update(ops []FederatedOperation, timeout time.Duration, onError func(FederatedOperation, error)) error {
```
Comment (wojtek-t) [inline federation/pkg/federation-controller/util/federated_updater.go]: s/onError/_/ (so that noone thinks that this is actually used).

**Comment by wojtek-t** [comment]
One minor thing.

Also - please fix tests.

File: federation/pkg/federation-controller/util/federated_updater.go (hunk 1 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/util/federated_updater.go]: Done.

**Comment by mwielgus** [comment]
Done.

**Comment by mwielgus** [comment]
Adding LGTM as agreed with @wojtek-t offline

**Pull request merged**